### ISSUE TITLE
Implement Kafka using BigQueue

### ISSUE THREAD

**Issue by mangalaman93** [grandecola/bigqueue#33]

**Comment by darkcoderrises** [comment]
Hi, I would like to work on this.

**Comment by cyucelen** [comment]
Is there any description about this usecase? What will be implemented exactly?

**Comment by mangalaman93** [comment]
Hi @cyucelen, this was just an idea that we could work on in future. Our current focus is to build Kafka like APIs for bigqueue which essentially allow multiple iterators over embedded bigqueue.

**Comment by cyucelen** [comment]
So you need to handle multiple readers concurrently for that feature right? I was using bigqueue for a while and waiting for multithreading support too (It's postponed to v0.5.0 as I see). I am not proficient in systems programming but i would like to help about it if you can show me some learning paths on my learning process. 

TLDR; Kafka like API got my attention and I would like to contribute

**Comment by mangalaman93** [comment]
Give me another week or so, we should have more fine grained issues created that you can help with. We are in the process of finalizing the kafka like APIs.

**Comment by cyucelen** [comment]
Take your time. It will educatory for me i guess.
